pkg/apis/workspaces/v1alpha1/projects.go

```diff
@@ -78,19 +68,37 @@ type CustomProjectSource struct {
 }
 
 type ZipProjectSource struct {
-	CommonProjectSource `json:",inline"`
+	// Zip project's source location address. Should be file path of the archive, e.g. file://$FILE_PATH
+	// +required
+	Location string `json:"location,omitempty"`
+
+	// Part of project to populate in the working directory.
+	// +optional
+	SparseCheckoutDir string `json:"sparseCheckoutDir,omitempty"`
 }
 
 type GitLikeProjectSource struct {
-	CommonProjectSource `json:",inline"`
+	// Part of project to populate in the working directory.
+	// +optional
+	SparseCheckoutDir string `json:"sparseCheckoutDir,omitempty"`
 
-	// The tag or commit id to reset the checked out branch to
+	// Defines from what the project should be checked out. Required if there are more than one remote configured
 	// +optional
-	StartPoint string `json:"startPoint,omitempty"`
+	CheckoutFrom *CheckoutFrom `json:"checkoutFrom,omitempty"`
+
+	// The remotes map which should be initialized in the git project. Must have at least one remote configured
+	// +required
+	Remotes map[string]string `json:"remotes,omitempty"`
```

Comment:
Why did you use `omitempty` *and* the `+required` command tag ?

Reply:
I'll remove `omitempty` since it's required.

Reply:
Ah, it must be optional in the schema, otherwise devfiles with a parent may be not valid.
So, I've made it optional.

Reply:
Ah, I see. So in order to support optional fields only in overrides, but still have them mandatory in the main definition, you would:
- Make it `optional` on the GO struct as you just did
- Still add a `required` constraint in the schema with [a transformation rule similar to this one](https://github.com/sleshchenko/devworkspace-api/blob/multipleRemotes/schema-transformation-rules/devworkspace/addRequiredFields.json#L23)

Reply:
In the future I plan to generate dedicated Go structures for overrides. But for now that the simplest way.

Reply:
It's done in https://github.com/devfile/api/pull/120/commits/8e6766364ef24192dfb8a936f549bd6560e94ea6